packages/server/src/model-view.ts

```diff
@@ -30,8 +31,18 @@ export function view<T>(
       return getModelState(model)
     }
     const newState = loadInput(getter)
-    state = newState
     isValid = true
+    if (opts?.compare) {
```

Comment:
Should we provide a default comparator function as a very basic optimisation, just like React does with diffing? For example, `opts.compare` is always optional and can be a function, if not provided, we do simple equality check?

If we don't want this, what is the default behavior for users and question, whether users are not going to end up defining this themselves all over the place?

Reply:
yes if `opts.compare` is `true` we check with reference equality, which is fine for _some_ views

the default behavior is to always emit the new value to the subscribers, unless compare is the comparison fn or is true for cheking reference equality

Eventually, we will probably want a higher-level `view` that does deep comparison, which might be called `comparedView` or something, which will be handy for component views that you don't want to re-send to the client if it doesn't change

Reply:
I guess that's when we introduce `memo` type of view! Shall we document this in some advanced section of the documentation?

Reply:
Yeah as a follow up we can do that. Before I was trying to use `lodash.isEqual` for the deep comparison, but it wasn't working for some reason. So in my app I have verified this memoization is helpful by using `JSON.stringify(a) === JSON.stringify(b)` which is obviously bad, but I don't know what was wrong with isEqual.